src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,66 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
```

Comment:
@trishume The value of the generic (rather than a `HandleRef` or similar) is that you can safely store the typed `EventedHandle<MyHandleType>` in tokio-core's `PollEvented`. We can then write a wrapper library that exposes e.g. `Channel` with operations that return `Future`s (where `Channel` is some wrapper around `PollEvented<EventedHandle<magenta::Channel>>`, similar to how TCP/UDP structures are implemented [in tokio-core](https://github.com/tokio-rs/tokio-core/blob/master/src/net/udp/mod.rs#L13)).